x/evm/keeper/statedb_test.go

```diff
@@ -936,41 +949,54 @@ func (suite *KeeperTestSuite) TestSetBalance() {
 }
 
 func (suite *KeeperTestSuite) TestDeleteAccount() {
+	suite.SetupTest()
+
 	supply := big.NewInt(100)
 	contractAddr := suite.DeployTestContract(suite.T(), suite.address, supply)
 
 	testCases := []struct {
-		name   string
-		addr   common.Address
-		expErr bool
+		name        string
+		addr        common.Address
+		expPass     bool
+		errContains string
 	}{
 		{
-			"remove address",
-			suite.address,
-			false,
+			name: "remove address",
+			addr: suite.address,
+			// TODO: check if this assumption is correct? It should not be possible to delete an "EthAccount" like that,
+			// only smart contracts?
+			errContains: "only smart contracts can be self-destructed",
```

Comment:
the `DeleteAccount` method's docstring says that only smart contracts should be deleted using this method - is that the case @facs95 @fedekunze? We never want to delete a `BaseAccount` using this, right?

Reply:
correct! afaik only `selfdestruct` opcode should call this